geps/gep-1619.md

```diff
@@ -163,43 +163,258 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
+`SameSite`, `HttpOnly`, and `Secure` as extended support, considering their partial acceptance among APIs. The
+`Partitioned` attribute lacks support in any implementation API and will be omitted. Similarly, since the `Expires`
+attribute is redundant compared to the preferred `Max-Age` attribute, it will also be skipped.
+
+```go
+// SessionPersistencePolicy provides a way to define session persistence rules
+// for a service.
+//
+// Support: Core
+type SessionPersistencePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    
+    // Spec defines the desired state of SessionPersistencePolicy.
+    Spec SessionPersistencePolicySpec `json:"spec"`
+    
+    // Status defines the current state of SessionPersistencePolicy.
+    Status SessionPersistencePolicyStatus `json:"status,omitempty"`
+}
+
+// SessionPersistencePolicySpec defines the desired state of
+// SessionPersistencePolicy.
+// Note: there is no Override or Default policy configuration.
+type SessionPersistencePolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // Services are the only valid API target references.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+    
+    // HttpCookie defines and configures a cookie to achieve
+    // session persistence. 
+    HttpCookie *HttpCookie `json:"httpCookie"`
+}
+
+// HttpCookie defines a cookie to achieve session persistence.
+//
+// Support: Core
+type HttpCookie struct {
+    // Name defines the cookie's name.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MaxLength=4096
+    Name String `json:"name,omitempty"`
+    
+    // Path defines the cookie's Path attribute.
+    //
+    // Support: Core
+    //
+    // +optional
+    // +kubebuilder:validation:MaxLength=4096
+    Path String `json:"path,omitempty"`
+    
+    // TTL defines the cookie's Max-Age attribute. 
+    // 
+    // Support: Core
+    //
+    // +optional
+    // +kubebuilder:validation:MaxLength=4096
+    TTL String `json:"ttl,omitempty"`
+
+    // SameSite defines the cookie's SameSite attribute.
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +kubebuilder:validation:Enum=Strict;Lax;None
+    SameSite SameSiteType `json:"sameSite,omitempty"`
+	
+    // Domain defines the cookie's Domain attribute.
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +kubebuilder:validation:MaxLength=4096
+    Domain String `json:"domain,omitempty"`
+    
+    // Secure defines the cookie's Secure attribute.
+    //
+    // Support: Extended
+    //
+    // +optional
+    Secure SecureMode `json:"secure,omitempty"`
+
+    // HttpOnly defines the cookie's HttpOnly attribute.
+    //
+    // Support: Extended
+    //
+    // +optional
+    HttpOnly HttpOnlyMode `json:"httpOnly,omitempty"`
+}
+
+// SameSiteType defines a cookie's SameSite attribute.
```

Comment:
Do we need to provide a security-by-default value for the Cookie attributes if people don't define them?

Reply:
I think the security-by-default approach is _ideal_ if that's something that all implementations can live with, but I think you're right that having any default may create problems. For now I say we write it up as security-by-default and check in if that's something any implementation has a significant problem with.

/cc @mlavacca @rainest @arkodg @sunjayBhatia @robscott

Reply:
Contour would most likely be ok with the proposal for secure defaults (e.g. I assume set Secure by default for cookies returned in responses to https requests) since we have a mechanism to rewrite the set-cookie header

Reply:
I suspect secure by default is what all implementations will choose. The question is if we want to provide an API to not be secure by default, and I think the answer is no.

Reply:
Just curious: say in case of a gRPC client that is copying the cookie, it is really up to the client application to honor (or not) the security attributes in the cookie, right? So if the application chooses not to honor them would you expect any problems (or should it fail as per this spec)?

Reply:
> Just curious: say in case of a gRPC client that is copying the cookie, it is really up to the client application to honor (or not) the security attributes in the cookie, right?

Per RFC https://datatracker.ietf.org/doc/html/rfc6265, I'd say yes.

> So if the application chooses not to honor them would you expect any problems (or should it fail as per this spec)?

Hmm interesting question. A user agent that does not abide by RFC6265, I would argue, is broke and non-compliant. I don't think Gateway API spec should mandate that implements check for non-compliant clients, seems like that should be covered by the standard in RFC6265.